### PR TITLE
Isolate voting dialog - Closes #723

### LINKED ISSUE #723: Isolate voteDialog from Voting component
### Expected behaviour
As per changes in the ticket #554 , we need to be able to open every dialog individually and isolated from any related component only by means of routing. To do so, the `VoteDialog` should have access to the list of delegates and vote delegates though `Store` .

### Actual behaviour
We pass the list of delegates though properties.

### PR DESCRIPTION
**Pull request by reyraa**
Replaces the lists of votes and unvotes with a dictionary including all votes together. This helps eliminate unnecessary iteration over delegates and vote lists to maintain vote status.  also, stores the list of delegates and votes in redux store to isolate the voting component from votingDialog component.
to do so, this PR:
 - Updates voting actions
   - Adds `votesFetched`, `delegatesFetched`, `voteToggled`
   - removes `removedFromVoteList ` and `addedToVoteList `
 - Refactors all components in voting and votingDialog to reflect the above changes.
 - Also moves Api calls from voting component to actions leaving the aforementioned component fully presentational.

Closes #723

### CODE REVIEW

File: src/components/voting/voting.js [hunk 1]
```diff
@@ -175,19 +110,22 @@ class Voting extends React.Component {
...
+          (!this.isInitial && this.props.delegates.length === 0) &&
+          <div className='hasPaddingRow empty-message'>No delegates found</div>
+        }
         <Waypoint bottomOffset='-80%'
-                  scrollableAncestor={window}
-                  key={this.state.delegates.length}
-                  onEnter={this.loadMore.bind(this)}></Waypoint>
+          scrollableAncestor={window}
+          key={this.props.delegates.length}
+          onEnter={this.loadMore.bind(this)}></Waypoint>
```
Comment (yasharAyari) [inline src/components/voting/voting.js]: `loadMore` function doesn't work when page scroller rich end of the page

File: src/utils/api/delegate.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ export const voteAutocomplete = (activePeer, username, votedList) => {
     listDelegates(activePeer, options)
     .then((response) => {
       resolve(response.delegates.filter(delegate =>
-        votedList.filter(item => item.username === delegate.username).length === 0,
+        Object.keys(votedList).filter(item => item === delegate.username).length === 0,
```
Comment (slaweet) [inline src/utils/api/delegate.js]: Please rename `votedList` since it is not a list anymore.

File: src/utils/api/delegate.js [hunk 3]
```diff
@@ -36,7 +36,10 @@ export const voteAutocomplete = (activePeer, username, votedList) => {
 
 export const unvoteAutocomplete = (username, votedList) =>
   new Promise((resolve) => {
-    resolve(votedList.filter(delegate => delegate.username.indexOf(username) !== -1));
+    resolve(
+      Object.keys(votedList)
+      .filter(delegate => delegate.indexOf(username) !== -1)
+      .map(element => ({ username: element, publicKey: votedList[element].publicKey })));
```
Comment (slaweet) [inline src/utils/api/delegate.js]: Please rename `votedList` since it is not a list anymore.

File: src/utils/api/delegate.test.js [hunk 4]
```diff
@@ -70,15 +70,19 @@ describe('Utils: Delegate', () => {
 
   describe('unvoteAutocomplete', () => {
     it('should return a promise', () => {
-      const votedList = ['genesis_1', 'genesis_2', 'genesis_3'];
+      const voteList = {
```
Comment (slaweet) [inline src/utils/api/delegate.test.js]: Please rename `votedList` since it is not a list anymore.

File: src/components/voting/voting.js (hunk 1 above)
Comment (reyraa) [reply src/components/voting/voting.js]: Done

**Pull request merged**